SignalServiceKit/src/Messages/OWSMessageDecrypter.m

```diff
@@ -0,0 +1,334 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSMessageDecrypter.h"
+#import "NSData+messagePadding.h"
+#import "NotificationsProtocol.h"
+#import "OWSAnalytics.h"
+#import "OWSBlockingManager.h"
+#import "OWSError.h"
+#import "OWSIdentityManager.h"
+#import "OWSSignalServiceProtos.pb.h"
+#import "TSAccountManager.h"
+#import "TSContactThread.h"
+#import "TSErrorMessage.h"
+#import "TSPreKeyManager.h"
+#import "TSStorageManager+PreKeyStore.h"
+#import "TSStorageManager+SessionStore.h"
+#import "TSStorageManager+SignedPreKeyStore.h"
+#import "TSStorageManager.h"
+#import "TextSecureKitEnv.h"
+#import <AxolotlKit/AxolotlExceptions.h>
+#import <AxolotlKit/SessionCipher.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSMessageDecrypter ()
+
+@property (nonatomic, readonly) TSStorageManager *storageManager;
+@property (nonatomic, readonly) YapDatabaseConnection *dbConnection;
+@property (nonatomic, readonly) OWSBlockingManager *blockingManager;
+@property (nonatomic, readonly) OWSIdentityManager *identityManager;
+
+@end
+
+#pragma mark -
+
+@implementation OWSMessageDecrypter
```

Comment:
I've pulled out the decryption logic from the message manager into a separate class.